src/ArrangerConduit.sol

```diff
@@ -0,0 +1,251 @@
+    /*** Router Functions                                                                       ***/
+    /**********************************************************************************************/
+
+    function deposit(bytes32 ilk, address asset, uint256 amount) external override {
+        // TODO: Use ERC20Helper
+        require(
+            ERC20Like(asset).transferFrom(msg.sender, address(this), amount),
+            "Conduit/deposit-transfer-failed"
+        );
+
+        positions[ilk][asset] += amount;
+        totalPositions[asset] += amount;
+    }
+
+    function withdraw(bytes32 ilk, address asset, address destination, uint256 withdrawAmount)
+        external override
+    {
+        // maxWithdraw < pendingWithdrawals < positions < totalWithdrawable < totalPositions
+        require(
+            withdrawAmount <= maxWithdraw[ilk][asset],
+            "Conduit/insufficient-withdrawal"
+        );
+
+        maxWithdraw[ilk][asset]        -= withdrawAmount;
+        pendingWithdrawals[ilk][asset] -= withdrawAmount;
+        positions[ilk][asset]          -= withdrawAmount;
+
+        totalPositions[asset]    -= withdrawAmount;
+        totalWithdrawable[asset] -= withdrawAmount;
+
+        require(
+            ERC20Like(asset).transfer(destination, withdrawAmount),
+            "Conduit/withdraw-transfer-failed"
+        );
+    }
+
+    function requestFunds(bytes32 ilk, address asset, uint256 amount, bytes memory data)
+        external override returns (uint256 fundRequestId)
+    {
+        fundRequestId = fundRequests[asset].length;  // Current length will be the next index
+
+        fundRequests[asset].push(FundRequest({
+            status:          StatusEnum.PENDING,
+            ilk:             ilk,
+            amountRequested: amount,
+            amountFilled:    0
+        }));
+
+        pendingWithdrawals[ilk][asset] += amount;
+
+        require(
+            pendingWithdrawals[ilk][asset] <= positions[ilk][asset],
+            "Conduit/insufficient-position"
+        );
+    }
+
+    function cancelFundRequest(address asset, uint256 fundRequestId) external override {
+        delete fundRequests[asset][fundRequestId];
+    }
+
+    /**********************************************************************************************/
+    /*** Fund Manager Functions                                                                 ***/
+    /**********************************************************************************************/
+
+    function drawFunds(address asset, uint256 amount) external override isFundManager {
+        outstandingPrincipal[asset] += amount;
+
+        require(
+            amount <= ERC20Like(asset).balanceOf(address(this)) - totalWithdrawable[asset],
+            "Conduit/insufficient-available-cash"
+        );
+
+        require(ERC20Like(asset).transfer(fundManager, amount), "Conduit/transfer-failed");
```

Comment:
Is there a case where we wouldn't want to have the operator also be the one who receives funds? I don't think this is the case, but will be good to double-check.